fishtest/utils/create_indexes.py

```diff
@@ -23,28 +23,62 @@
 pgns = db['pgns']
 
 
-indexList = { 'runs' : [ [(u'args.username', ASCENDING)],
-                         [(u'finished', ASCENDING), (u'last_updated', DESCENDING), (u'start_time', DESCENDING)] ],
-                         # [(u'finished', ASCENDING), (u'last_updated', DESCENDING)],
-                         # [(u'deleted', ASCENDING)],
-                         # [(u'deleted', ASCENDING), (u'finished', ASCENDING)],
-                         # [(u'last_updated', DESCENDING), (u'start_time', DESCENDING)],
-                         # [(u'tasks.pending', ASCENDING)],
-              'pgns' : [ [('run_id', ASCENDING)] ],
-              'users' : [ [('username', ASCENDING)] ],
-              'actions' : [ [('username', ASCENDING)] ],
-            }
-uniqueList = [ 'username' ]
+index_list = {
+  'runs' : [
+    [
+      # Index used for querying combos of username / Greens / LTC
+      ('finished', DESCENDING),
+      ('args.username', ASCENDING),
+      ('args.ltc', ASCENDING),
+      ('results_info.style', ASCENDING),
+      ('last_updated', DESCENDING),
+    ],
+    [
+      # Index used for get_unfinished_runs()
+      ('finished', ASCENDING),
+      ('last_updated', DESCENDING),
+      ('start_time', DESCENDING),
+    ],
+    # [(u'deleted', ASCENDING)],
+    # [(u'deleted', ASCENDING), (u'finished', ASCENDING)],
+    # [(u'last_updated', DESCENDING), (u'start_time', DESCENDING)],
+    # [(u'tasks.pending', ASCENDING)],
+  ],
+  'old_runs' : [
+    [
+      # Index used for querying combos of username / Greens / LTC only
+      ('finished', DESCENDING),
+      ('args.username', ASCENDING),
+      ('args.ltc', ASCENDING),
+      ('results_info.style', ASCENDING),
+      ('_id', DESCENDING),
+    ],
+  ],
+  'pgns' : [
+    [('run_id', ASCENDING)]
+  ],
+  'users' : [
+    [('username', ASCENDING)]
+  ],
+  'actions' : [
+    [
+      ('action', ASCENDING),
+      ('username', ASCENDING),
+      ('_id', DESCENDING)
```

Comment:
Index on `_id` is always present by  default

Reply:
same as https://github.com/glinscott/fishtest/pull/570#discussion_r406834968. `_id` needs to be in this compound index since the actions queries all sort by `_id` after filtering.

Example code:
```python
from pprint import pprint
from pymongo import MongoClient, DESCENDING

def print_query_plan(mongo_cursor):
    explanation = mongo_cursor.explain()
    winning_plan = explanation['queryPlanner']['winningPlan'];
    if winning_plan:
        # Query starts with an index scan
        pprint(winning_plan)
    else:
        # Slow query - collection scan
        pprint(explanation['queryPlanner'])

conn = MongoClient()
print_query_plan(conn['fishtest_new']['actions'].find({
  'action': 'updated_stats',
  'username': 'username'
}, sort=[('_id', DESCENDING)]))
```

With `_id` in the index this fetches entries already-sorted using an index scan (fast):
```
{'inputStage': {'direction': 'forward',
                'indexBounds': {'_id': ['[MaxKey, MinKey]'],
                                'action': ['["updated_stats", '
                                           '"updated_stats"]'],
                                'username': ['["username", "username"]']},
                'indexName': 'action_1_username_1__id_-1',
                'indexVersion': 2,
                'isMultiKey': False,
                'isPartial': False,
                'isSparse': False,
                'isUnique': False,
                'keyPattern': {'_id': -1, 'action': 1, 'username': 1},
                'multiKeyPaths': {'_id': [], 'action': [], 'username': []},
                'stage': 'IXSCAN'},
 'stage': 'FETCH'}
```

Without `_id` in the index this fetches entries first, then sorts them separately (slow):
```
{'inputStage': {'inputStage': {'inputStage': {'direction': 'forward',
                                              'indexBounds': {'action': ['["updated_stats", '
                                                                         '"updated_stats"]'],
                                                              'username': ['["username", '
                                                                           '"username"]']},
                                              'indexName': 'action_1_username_1',
                                              'indexVersion': 2,
                                              'isMultiKey': False,
                                              'isPartial': False,
                                              'isSparse': False,
                                              'isUnique': False,
                                              'keyPattern': {'action': 1,
                                                             'username': 1},
                                              'multiKeyPaths': {'action': [],
                                                                'username': []},
                                              'stage': 'IXSCAN'},
                               'stage': 'FETCH'},
                'stage': 'SORT_KEY_GENERATOR'},
 'sortPattern': {'_id': -1},
 'stage': 'SORT'}
```